### PR TITLE
docs: Enables Docusaurus versioning for previous version of ICS docs

### PR DESCRIPTION
**Pull request by mmulji-ic**
## Description

Part of docs Epic https://github.com/cosmos/gaia/issues/2309.

This PR enables multiple versions of the Interchain Security docs to be displayed upon deployment.
Note: Limited the number of docs, as previous versions of docs were raw markdown and did not have a Docusaurus installation.

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [x] included the correct `docs:` prefix in the PR title
- [x] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/interchain-security/blob/main/CONTRIBUTING.md#pr-targeting))
- [x] provided a link to the relevant issue or specification
- [x] reviewed "Files changed" and left comments if necessary 
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] Confirmed the correct `docs:` prefix in the PR title
- [ ] Confirmed all author checklist items have been addressed 
- [ ] Confirmed that this PR only changes documentation
- [ ] Reviewed content for consistency
- [ ] Reviewed content for spelling and grammar
- [ ] Tested instructions (if applicable)
- [ ] Checked that the documentation website can be built and deployed successfully (run `make build-docs`)

### CODE REVIEW

File: docs/versions.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+[
+    "v3.1.0",
```
Comment (p-offtermatt) [inline docs/versions.json:2]: Could you add a reminder to add new releases here to https://github.com/cosmos/interchain-security/blob/main/RELEASE_PROCESS.md?

**Review by p-offtermatt — CHANGES_REQUESTED** [review]
Left one comment. Also, the versioning doesn't work locally for me - v3.1.0 is not selectable

**Comment by p-offtermatt** [comment]
This is still broken for me - the dropdown under the version has both Next and v3.1.0.
They are the same, appearantly, and clicking v3.1.0 just brings me to Next.
It's weird, because it seems like clicking v3.1.0 when currently on Next is broken, as it does nothing.


This is not what I'd exect - I'd expect v3.1.0 to not be listed at all if it only redirects to Next anyways, and just have Next and old versions,
i.e. see the sdk docs - https://docs.cosmos.network/main/learn. Every version is clickable from every other version.

**Comment by mmulji-ic** [comment]
After the Docusaurus deps bump, seems only main and release/v4.0.x branches are compatible content wise. 
Alas, Docusaurus doesn't like any release/* branches either. Therefore, will leave this PR open, until we have another compatible branch to test. I don't want to updates deps in previous release branches.

**Comment by mmulji-ic** [comment]
@MSalopek has another version under review

**Pull request closed without merging**